go-controller/pkg/clustermanager/userdefinednetwork/controller.go

```diff
@@ -169,8 +171,32 @@ func (c *Controller) ReconcileNetAttachDef(key string) error {
 
 // ReconcileNamespace enqueue relevant Cluster UDN CR requests following namespace events.
 func (c *Controller) ReconcileNamespace(key string) error {
-	// TODO: reconcile affected namespaced only.
-	c.cudnController.Reconcile(key)
+	c.namespaceTrackerLock.Lock()
```

Comment:
should we use rlock here considering namespace updates may be more often than udn updates?

Reply:
Thanks for the heads up, done.